src/arrayMethodSort.js

```diff
@@ -4,8 +4,41 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  const defaultCompareFunction = (a, b) => {
+    const x = String(a);
+    const y = String(b);
+
+    if (x > y) {
+      return 1;
+    }
+
+    if (x < y) {
+      return -1;
+    }
+
+    return 0;
+  };
+
+  [].__proto__.sort2 = function (compareFunction) {
+    const arr = this;
+
+    const compareFn =
+      typeof compareFunction === 'function'
+        ? compareFunction
+        : defaultCompareFunction;
+
+    for (let i = 0; i < arr.length - 1; i++) {
+      for (let j = 0; j < arr.length - 1 - i; j++) {
+        if (compareFn(arr[j], arr[j + 1]) > 0) {
+          const temp = arr[j];
+
+          arr[j] = arr[j + 1];
+          arr[j + 1] = temp;
+        }
+      }
+    }
```

Comment:
In the bubble sort implementation, you can improve performance slightly by checking if no elements were swapped in the inner loop. If no swaps occurred, the array is already sorted, and you can break out of the loops early to avoid unnecessary iterations.